crates/oxvg_ast/src/implementations/markup5ever.rs

```diff
@@ -1398,14 +1394,8 @@ impl Node for Element5Ever {
     }
 
     #[allow(refining_impl_trait)]
-    fn set_parent_node(&self, new_parent: &impl Node<Atom = Self::Atom>) -> Option<Element5Ever> {
-        let new_parent_element = new_parent as &dyn std::any::Any;
-        let new_parent_element = new_parent_element.downcast_ref::<Element5Ever>().unwrap();
-        let old_parent = Element5Ever {
-            node: self.node.set_parent_node(&new_parent_element.node)?,
-            selector_flags: Cell::new(None),
-        };
-        Some(old_parent)
+    fn set_parent_node(&self, new_parent: &Self::Parent) -> Option<Node5Ever> {
+        self.node.set_parent_node(&new_parent)
```

Comment:
```suggestion
        self.node.set_parent_node(&new_parent).map(|node| Element5Ever { node, selector_flags: Cell::new(None) })
```
Ideally would return `Element5Ever` since all element parents should be an element.
(Note, document nodes can also be used in `Element5Ever`)